File: dotnet-zero-engine/Dto/ConsoleMessageDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DotnetZero.Engine.Dto
{
    public class ConsoleMessageDto
    {
        public string Message { get; set; }
        public ConsoleMessageType Type { get; set; }
        public int? Line { get; set; }
        public int? Column { get; set; }
        public string FilePath { get; set; }
        public string Code { get; set; }
    }
}

File: dotnet-zero-ui/src/app/dto/console-message-dto.ts
export interface ConsoleMessageDto {
  message: string;
  type: ConsoleMessageType;
  code?: string;
  filePath?: string;
  line?: number;
  column?: number;
}

export enum ConsoleMessageType {
  Hidden = 0,
  //
  // Summary:
  //     Information that does not indicate a problem (i.e. not prescriptive).
  Info = 1,
  //
  // Summary:
  //     Something suspicious but allowed.
  Warning = 2,
  //
  // Summary:
  //     Something not allowed by the rules of the language or other authority.
  Error = 3
}
